src/builtins.cpp

```diff
@@ -419,62 +419,81 @@ bool lStartsWithLLVM(llvm::StringRef name) { return name.starts_with("llvm."); }
-
-    // {ISPCTarget::avx2vnni_i32x8, ISPCTarget::avx2_i32x8},
-    // {ISPCTarget::avx2_i32x8, ISPCTarget::avx1_i32x8},
-    // {ISPCTarget::avx1_i32x8, ISPCTarget::sse4_i32x8},
-    // {ISPCTarget::sse4_i32x8, ISPCTarget::sse2_i32x8},
-    // {ISPCTarget::sse2_i32x8, ISPCTarget::generic_i32x8},
-
-    // {ISPCTarget::avx2vnni_i32x16, ISPCTarget::avx2_i32x16},
-    // {ISPCTarget::avx2_i32x16, ISPCTarget::avx1_i32x16},
-    // {ISPCTarget::avx1_i32x16, ISPCTarget::generic_i32x16},
-
-    // {ISPCTarget::avx2_i16x16, ISPCTarget::sse4_i16x16},
-    // {ISPCTarget::sse4_i16x16, ISPCTarget::generic_i16x16},
-
-    // {ISPCTarget::sse4_i8x16, ISPCTarget::generic_i8x16},
-
-    // {ISPCTarget::avx2_i8x32, ISPCTarget::generic_i8x32},
+    //{ISPCTarget::avx512skx_x64, ISPCTarget::generic_i1x64},
+
+    {ISPCTarget::avx2vnni_i32x4, ISPCTarget::avx2_i32x4},
+    {ISPCTarget::avx2_i32x4, ISPCTarget::avx1_i32x4},
+    {ISPCTarget::avx1_i32x4, ISPCTarget::sse4_i32x4},
+    {ISPCTarget::sse4_i32x4, ISPCTarget::sse41_i32x4},
+    {ISPCTarget::sse41_i32x4, ISPCTarget::sse2_i32x4},
+    {ISPCTarget::sse2_i32x4, ISPCTarget::generic_i32x4},
+
+    {ISPCTarget::avx2vnni_i32x8, ISPCTarget::avx2_i32x8},
+    {ISPCTarget::avx2_i32x8, ISPCTarget::avx1_i32x8},
+    {ISPCTarget::avx1_i32x8, ISPCTarget::sse4_i32x8},
+    {ISPCTarget::sse4_i32x8, ISPCTarget::sse41_i32x8},
+    {ISPCTarget::sse41_i32x8, ISPCTarget::sse2_i32x8},
+    {ISPCTarget::sse2_i32x8, ISPCTarget::generic_i32x8},
+
+    {ISPCTarget::avx2vnni_i32x16, ISPCTarget::avx2_i32x16},
+    {ISPCTarget::avx2_i32x16, ISPCTarget::avx1_i32x16},
+    {ISPCTarget::avx1_i32x16, ISPCTarget::generic_i32x16},
+
+    {ISPCTarget::avx2_i64x4, ISPCTarget::avx1_i64x4},
+    {ISPCTarget::avx1_i64x4, ISPCTarget::generic_i64x4},
+
+    {ISPCTarget::avx2_i16x16, ISPCTarget::generic_i16x16},
+
+    {ISPCTarget::avx2_i8x32, ISPCTarget::generic_i8x32},
+
+    {ISPCTarget::sse4_i8x16, ISPCTarget::sse41_i8x16},
+    {ISPCTarget::sse41_i8x16, ISPCTarget::generic_i8x16},
+
+    {ISPCTarget::sse4_i16x8, ISPCTarget::sse41_i16x8},
+    {ISPCTarget::sse41_i16x8, ISPCTarget::generic_i16x8},
+
+    {ISPCTarget::wasm_i32x4, ISPCTarget::generic_i32x4},
+
+    {ISPCTarget::gen9_x8, ISPCTarget::generic_i1x8},
+    {ISPCTarget::gen9_x16, ISPCTarget::generic_i1x16},
+    {ISPCTarget::xelp_x8, ISPCTarget::generic_i1x8},
+    {ISPCTarget::xelp_x16, ISPCTarget::generic_i1x16},
+    {ISPCTarget::xehpg_x8, ISPCTarget::generic_i1x8},
+    {ISPCTarget::xehpg_x16, ISPCTarget::generic_i1x16},
+    {ISPCTarget::xehpc_x16, ISPCTarget::generic_i1x16},
+    {ISPCTarget::xehpc_x32, ISPCTarget::generic_i1x32},
+    {ISPCTarget::xelpg_x8, ISPCTarget::generic_i1x8},
+    {ISPCTarget::xelpg_x16, ISPCTarget::generic_i1x16},
+    {ISPCTarget::xe2hpg_x16, ISPCTarget::generic_i1x16},
+    {ISPCTarget::xe2hpg_x32, ISPCTarget::generic_i1x32},
+    {ISPCTarget::xe2lpg_x16, ISPCTarget::generic_i1x16},
+    {ISPCTarget::xe2lpg_x32, ISPCTarget::generic_i1x32}
```

Comment:
If we want to use generic targets for `xe`, then we need to support building them for `xe` arch in this file: https://github.com/ispc/ispc/blob/main/cmake/GenericTargets.cmake.

Reply:
I decided to limit the change for x86 targets only since on other targets I see many issues that require careful investigation. Right now I don't want to do this, and prefer to enable x86 targets only for further avx10 support.

Reply:
Full testing is here: https://github.com/aneshlya/ispc/actions/runs/13709157474